### PR TITLE
Initialized locale in context

### LINKED ISSUE #33374: Blockwislist v3.0 : Call to a member function formatNumber() on null
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

After added a product to a wishlist, I want to refresh statistics.

image

### Expected behavior

_No response_

### Steps to reproduce

1. Go to BO
2. Install `blockwishlist` (v3) 
3. Go to configure page
4. Go to Statistics tab
5. There is no statistics
6. Go to FO
7. Connect as John Doe for example
8. Add some products to the basic wishlist "My wishlist"
9. Logout from John Doe
10. Return to the BO
11. On the Statistics tab, click on the refresh Tab...
12. :boom: 

#### Acceptance Test

Unksip the test : https://github.com/PrestaShop/PrestaShop/blob/34aa6b54b6d4f7ddd9b91c02cca495879bfe5ab0/tests/UI/campaigns/modules/blockwishlist/02_configuration/02_statisticsTabSettings.ts#L164-L165

### PrestaShop version(s) where the bug happened

9.0.x

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

blockwishlist v3.0

### Your company or customer's name goes here (if applicable).

@PrestaShopCorp

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | The locale was not initialized correctly 
| Type?             | bug fix 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See https://github.com/PrestaShop/PrestaShop/issues/33374
| UI Tests          | https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/11052737822
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/33374
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Comment by kpodemski** [comment]
@nicosomb cs fixer 👍🏻

**Review by boherm — DISMISSED** [review]

File: src/Core/Context/LanguageContextBuilder.php [hunk 1]
```diff
@@ -121,5 +126,6 @@ public function buildLegacyContext(): void
         $this->assertArguments();
         $language = $this->objectModelLanguageRepository->get(new LanguageId($this->languageId));
         $this->contextStateManager->setLanguage($language);
+        $this->contextStateManager->setCurrentLocale($this->getLocaleByLanguage($language));
```
Comment (jolelievre) [inline src/Core/Context/LanguageContextBuilder.php:129]: All good but we need to update the related tests in `Tests\Unit\Core\Context\LanguageContextBuilderTest::testBuildLegacyContext`

We need to assert that when `buildLegacyContext` is called the ContextStateManager is also called to override the locale

File: tests/Unit/Core/Context/LanguageContextBuilderTest.php [hunk 2]
```diff
@@ -89,6 +89,7 @@ public function testBuildLegacyContext(): void
         $builder->setLanguageId($objectModelLanguage->getId());
 
         $contextManagerMock->expects(static::once())->method('setLanguage')->with($objectModelLanguage);
+        $contextManagerMock->expects(static::once())->method('setCurrentLocale')->with($this->mockObjectModelLanguageRepository($objectModelLanguage));
```
Comment (jolelievre) [inline tests/Unit/Core/Context/LanguageContextBuilderTest.php]: I think you need to mock the language object only once and store it in a variable, you can then pass on this line and on line 87

**Review by jolelievre — APPROVED** [review]
Thanks @nicosomb

**Review by boherm — APPROVED** [review]

**Review by tleon — APPROVED** [review]

**Comment by paulnoelcholot** [comment]
Hello @nicosomb,

I tested your PR step by step and I found this at the step 12 : 

image

**Comment by nicosomb** [comment]
This bug is fixed here https://github.com/PrestaShop/blockwishlist/pull/280 where you told that you have an issue... which is fixed in this PR :)

**Comment by boherm** [comment]
Tested on my side! It's all ok, thanks @nicosomb!

**Pull request merged**